### PR TITLE
python312Packages.torchbench: 0.0.31

### PR DESCRIPTION
**Pull request by drupol**
## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by drupol** [comment]
@natsukium Can you help me to fix `torchbench` in this PR ?

File: pkgs/development/python-modules/monkeytype/default.nix [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+
+buildPythonPackage rec {
+  pname = "monkeytype";
+  version = "23.3.0";
+  pyproject = true;
+
+  src = fetchFromGitHub {
+    owner = "Instagram";
+    repo = "MonkeyType";
+    rev = "v${version}";
```
Comment (GaetanLepage) [inline pkgs/development/python-modules/monkeytype/default.nix:19]:
```suggestion
    rev = "refs/tags/v${version}";
```

File: pkgs/development/python-modules/monkeytype/default.nix [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+    hash = "sha256-DQ/3go53+0PQkhZcL2dX8MI/z4Iq7kTYd5EbacMNxT4=";
+  };
+
+  build-system = [
+    setuptools
+    wheel
```
Comment (GaetanLepage) [inline pkgs/development/python-modules/monkeytype/default.nix:25]:
```suggestion
```
Added automatically by `nix-init` but not needed.
-> Same for the other two packages.

File: pkgs/development/python-modules/monkeytype/default.nix [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+  dependencies = [
+    libcst
+    mypy-extensions
+  ];
+
+  pythonImportsCheck = [
+    "monkeytype"
+  ];
+
+  meta = {
```
Comment (GaetanLepage) [inline pkgs/development/python-modules/monkeytype/default.nix:37]:
```suggestion
  ];

  nativeCheckInputs = [
    pytestCheckHook
  ];

  meta = {
```

Can you try enabling the tests please ?

-> Same for the other two packages.

File: pkgs/development/python-modules/submitit/default.nix [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+  ];
+
+  pythonImportsCheck = [
+    "submitit"
+  ];
+
+  meta = {
+    description = "Python 3.8+ toolbox for submitting jobs to Slurm";
+    homepage = "https://github.com/facebookincubator/submitit";
+    license = lib.licenses.mit;
```
Comment (GaetanLepage) [inline pkgs/development/python-modules/submitit/default.nix:40]:
```suggestion
    homepage = "https://github.com/facebookincubator/submitit";
    changelog = "https://github.com/facebookincubator/submitit/releases/tag/${version}";
    license = lib.licenses.mit;
```

Same for `torchbench`.

**Review by GaetanLepage — CHANGES_REQUESTED** [review]

**Comment by drupol** [comment]
I fixed all the comments, ready for another round of review. I'm still looking for fixing `torchbench` too.

File: pkgs/development/python-modules/submitit/default.nix [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+  fetchFromGitHub,
+  setuptools,
+  cloudpickle,
+  typing-extensions,
+  pytestCheckHook,
+}:
+
+buildPythonPackage rec {
+  pname = "submitit";
+  version = "1.2.0";
```
Comment (natsukium) [inline pkgs/development/python-modules/submitit/default.nix:13]: This release is three years old. Is there any reason not to use the latest 1.5.2?

File: pkgs/development/python-modules/torchbench/default.nix [hunk 6]
```diff
@@ -0,0 +1,76 @@
...
+  pytest-benchmark,
+  pyyaml,
+  requests,
+  submitit,
+  tabulate,
+  transformers,
+}:
+
+buildPythonPackage {
+  pname = "torchbench";
```
Comment (natsukium) [inline pkgs/development/python-modules/torchbench/default.nix:28]: `torchbench` is [another project](https://pypi.org/project/torchbench/). How about `torch-bench`?

File: pkgs/development/python-modules/torchbench/default.nix [hunk 7]
```diff
@@ -0,0 +1,76 @@
...
+  version = "0-unstable-2024-10-17";
+  format = "setuptools";
```
Comment (natsukium) [inline pkgs/development/python-modules/torchbench/default.nix:30]:
```suggestion
  pyproject = true;
```

File: pkgs/development/python-modules/torchbench/default.nix [hunk 8]
```diff
@@ -0,0 +1,76 @@
...
+
+  src = fetchFromGitHub {
+    owner = "pytorch";
+    repo = "benchmark";
+    rev = "00c9b9ed61b70425e6875e0c4d6c5ad18a1b1d4c";
+    hash = "sha256-Sc50BvIQWVZgotOOC1sG3qjVHRbqxDcLH4xA+Vt/T3o=";
+    fetchSubmodules = true;
+  };
+
```
Comment (natsukium) [inline pkgs/development/python-modules/torchbench/default.nix:39]:
```suggestion
  postPatch = ''
    substituteInPlace torchbenchmark/__init__.py \
      --replace-fail "REPO_PATH = Path(os.path.abspath(__file__)).parent.parent" "REPO_PATH = Path('$src')"
  '';
```
A relative path outside the package called `REPO_PATH` is used in many places. We could replace all REPO_PATHs in this way.
I'm not sure that's the best approach.

https://github.com/search?q=repo%3Apytorch%2Fbenchmark%20REPO_PATH&type=code

File: pkgs/development/python-modules/torchbench/default.nix [hunk 9]
```diff
@@ -0,0 +1,76 @@
...
+    tabulate
+    transformers
+  ];
+
+  pythonImportsCheck = [
+    "torchbenchmark"
+  ];
+
+  meta = {
+    description = "TorchBench is a collection of open source benchmarks used to evaluate PyTorch performance";
```
Comment (natsukium) [inline pkgs/development/python-modules/torchbench/default.nix:71]:
```suggestion
    description = "Collection of open source benchmarks used to evaluate PyTorch performance";
```

File: pkgs/development/python-modules/torchbench/default.nix (hunk 6 above)
Comment (drupol) [reply pkgs/development/python-modules/torchbench/default.nix:28]: This is precisely that project https://pypi.org/project/torchbench/ that I would like to package here. How to make sure the project I see on Pypi is the one on Github?

Comment (GaetanLepage) [reply pkgs/development/python-modules/torchbench/default.nix:28]: - [Pypi's `torchbench`](https://pypi.org/project/torchbench) is [paperswithcode/torchbench](https://github.com/paperswithcode/torchbench)
- [Pypi's `torchbenchmark`](https://pypi.org/project/torchbenchmark/) is [pytorch/benchmark](https://github.com/pytorch/benchmark)

Comment (GaetanLepage) [reply pkgs/development/python-modules/torchbench/default.nix:28]: If you would like to package `torchbench` (the first one), you have to use `fetchPypi` as they unfortunarely [don't tag released on GitHub](https://github.com/paperswithcode/torchbench/tags).

**Comment by GaetanLepage** [comment]
CI complains that `top-level/python-packages.nix` is not formatted... I am not sure why it triggered in this PR...

**Comment by drupol** [comment]
I think I fixed all the feedback I received.

In the end, after fixing the origin of the `torchbench` package based on your feedback, it turns out that `submitit` and `monkeytype` are not needed in it. I took the decision to leave them in the PR, they might be useful for someone else at some point.

**Comment by GaetanLepage** [comment]
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 349472`

---
### `x86_64-linux`
<details>
  <summary>:white_check_mark: 16 packages built:</summary>
  <ul>
    <li>python311Packages.monkeytype</li>
    <li>python311Packages.monkeytype.dist</li>
    <li>python311Packages.sotabenchapi</li>
    <li>python311Packages.sotabenchapi.dist</li>
    <li>python311Packages.submitit</li>
    <li>python311Packages.submitit.dist</li>
    <li>python311Packages.torchbench</li>
    <li>python311Packages.torchbench.dist</li>
    <li>python312Packages.monkeytype</li>
    <li>python312Packages.monkeytype.dist</li>
    <li>python312Packages.sotabenchapi</li>
    <li>python312Packages.sotabenchapi.dist</li>
    <li>python312Packages.submitit</li>
    <li>python312Packages.submitit.dist</li>
    <li>python312Packages.torchbench</li>
    <li>python312Packages.torchbench.dist</li>
  </ul>
</details>

---
### `aarch64-linux`
<details>
  <summary>:white_check_mark: 16 packages built:</summary>
  <ul>
    <li>python311Packages.monkeytype</li>
    <li>python311Packages.monkeytype.dist</li>
    <li>python311Packages.sotabenchapi</li>
    <li>python311Packages.sotabenchapi.dist</li>
    <li>python311Packages.submitit</li>
    <li>python311Packages.submitit.dist</li>
    <li>python311Packages.torchbench</li>
    <li>python311Packages.torchbench.dist</li>
    <li>python312Packages.monkeytype</li>
    <li>python312Packages.monkeytype.dist</li>
    <li>python312Packages.sotabenchapi</li>
    <li>python312Packages.sotabenchapi.dist</li>
    <li>python312Packages.submitit</li>
    <li>python312Packages.submitit.dist</li>
    <li>python312Packages.torchbench</li>
    <li>python312Packages.torchbench.dist</li>
  </ul>
</details>

---
### `x86_64-darwin`
<details>
  <summary>:white_check_mark: 16 packages built:</summary>
  <ul>
    <li>python311Packages.monkeytype</li>
    <li>python311Packages.monkeytype.dist</li>
    <li>python311Packages.sotabenchapi</li>
    <li>python311Packages.sotabenchapi.dist</li>
    <li>python311Packages.submitit</li>
    <li>python311Packages.submitit.dist</li>
    <li>python311Packages.torchbench</li>
    <li>python311Packages.torchbench.dist</li>
    <li>python312Packages.monkeytype</li>
    <li>python312Packages.monkeytype.dist</li>
    <li>python312Packages.sotabenchapi</li>
    <li>python312Packages.sotabenchapi.dist</li>
    <li>python312Packages.submitit</li>
    <li>python312Packages.submitit.dist</li>
    <li>python312Packages.torchbench</li>
    <li>python312Packages.torchbench.dist</li>
  </ul>
</details>

---
### `aarch64-darwin`
<details>
  <summary>:white_check_mark: 16 packages built:</summary>
  <ul>
    <li>python311Packages.monkeytype</li>
    <li>python311Packages.monkeytype.dist</li>
    <li>python311Packages.sotabenchapi</li>
    <li>python311Packages.sotabenchapi.dist</li>
    <li>python311Packages.submitit</li>
    <li>python311Packages.submitit.dist</li>
    <li>python311Packages.torchbench</li>
    <li>python311Packages.torchbench.dist</li>
    <li>python312Packages.monkeytype</li>
    <li>python312Packages.monkeytype.dist</li>
    <li>python312Packages.sotabenchapi</li>
    <li>python312Packages.sotabenchapi.dist</li>
    <li>python312Packages.submitit</li>
    <li>python312Packages.submitit.dist</li>
    <li>python312Packages.torchbench</li>
    <li>python312Packages.torchbench.dist</li>
  </ul>
</details>

File: pkgs/development/python-modules/sotabenchapi/default.nix [hunk 10]
```diff
@@ -0,0 +1,51 @@
...
+  format = "setuptools";
+
+  src = fetchPypi {
+    inherit pname version;
+    hash = "sha256-thbVH4aLmEgi8K17PkmbUg4nHqGj+dEiXPDILjvQMzk=";
+  };
+
+  postPatch = ''
+    touch requirements.txt
+  '';
```
Comment (GaetanLepage) [inline pkgs/development/python-modules/sotabenchapi/default.nix:26]: You removed my comments ?

Comment (GaetanLepage) [reply pkgs/development/python-modules/sotabenchapi/default.nix:26]: Oh no, this is another file. The `requirements.txt` is missing too ?

**Comment by drupol** [comment]
@ofborg build python312Packages.torchbench python312Packages.sotabenchapi python312Packages.monkeytype python312Packages.submitit

**Comment by drupol** [comment]
@ofborg build python312Packages.torchbench python312Packages.sotabenchapi python312Packages.monkeytype python312Packages.submitit

File: pkgs/development/python-modules/sotabenchapi/default.nix (hunk 10 above)
Comment (GaetanLepage) [reply pkgs/development/python-modules/sotabenchapi/default.nix:26]: Maybe add a comment too. The reason for this patch is not obvious when you look at it.

Comment (drupol) [reply pkgs/development/python-modules/sotabenchapi/default.nix:26]: Will do now.

**Comment by drupol** [comment]
@ofborg build python312Packages.torchbench python312Packages.sotabenchapi python312Packages.monkeytype python312Packages.submitit

**Comment by GaetanLepage** [comment]
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 349472`

---
### `x86_64-linux`
<details>
  <summary>:white_check_mark: 16 packages built:</summary>
  <ul>
    <li>python311Packages.monkeytype</li>
    <li>python311Packages.monkeytype.dist</li>
    <li>python311Packages.sotabenchapi</li>
    <li>python311Packages.sotabenchapi.dist</li>
    <li>python311Packages.submitit</li>
    <li>python311Packages.submitit.dist</li>
    <li>python311Packages.torchbench</li>
    <li>python311Packages.torchbench.dist</li>
    <li>python312Packages.monkeytype</li>
    <li>python312Packages.monkeytype.dist</li>
    <li>python312Packages.sotabenchapi</li>
    <li>python312Packages.sotabenchapi.dist</li>
    <li>python312Packages.submitit</li>
    <li>python312Packages.submitit.dist</li>
    <li>python312Packages.torchbench</li>
    <li>python312Packages.torchbench.dist</li>
  </ul>
</details>

---
### `aarch64-linux`
<details>
  <summary>:white_check_mark: 16 packages built:</summary>
  <ul>
    <li>python311Packages.monkeytype</li>
    <li>python311Packages.monkeytype.dist</li>
    <li>python311Packages.sotabenchapi</li>
    <li>python311Packages.sotabenchapi.dist</li>
    <li>python311Packages.submitit</li>
    <li>python311Packages.submitit.dist</li>
    <li>python311Packages.torchbench</li>
    <li>python311Packages.torchbench.dist</li>
    <li>python312Packages.monkeytype</li>
    <li>python312Packages.monkeytype.dist</li>
    <li>python312Packages.sotabenchapi</li>
    <li>python312Packages.sotabenchapi.dist</li>
    <li>python312Packages.submitit</li>
    <li>python312Packages.submitit.dist</li>
    <li>python312Packages.torchbench</li>
    <li>python312Packages.torchbench.dist</li>
  </ul>
</details>

---
### `x86_64-darwin`
<details>
  <summary>:white_check_mark: 16 packages built:</summary>
  <ul>
    <li>python311Packages.monkeytype</li>
    <li>python311Packages.monkeytype.dist</li>
    <li>python311Packages.sotabenchapi</li>
    <li>python311Packages.sotabenchapi.dist</li>
    <li>python311Packages.submitit</li>
    <li>python311Packages.submitit.dist</li>
    <li>python311Packages.torchbench</li>
    <li>python311Packages.torchbench.dist</li>
    <li>python312Packages.monkeytype</li>
    <li>python312Packages.monkeytype.dist</li>
    <li>python312Packages.sotabenchapi</li>
    <li>python312Packages.sotabenchapi.dist</li>
    <li>python312Packages.submitit</li>
    <li>python312Packages.submitit.dist</li>
    <li>python312Packages.torchbench</li>
    <li>python312Packages.torchbench.dist</li>
  </ul>
</details>

---
### `aarch64-darwin`
<details>
  <summary>:white_check_mark: 16 packages built:</summary>
  <ul>
    <li>python311Packages.monkeytype</li>
    <li>python311Packages.monkeytype.dist</li>
    <li>python311Packages.sotabenchapi</li>
    <li>python311Packages.sotabenchapi.dist</li>
    <li>python311Packages.submitit</li>
    <li>python311Packages.submitit.dist</li>
    <li>python311Packages.torchbench</li>
    <li>python311Packages.torchbench.dist</li>
    <li>python312Packages.monkeytype</li>
    <li>python312Packages.monkeytype.dist</li>
    <li>python312Packages.sotabenchapi</li>
    <li>python312Packages.sotabenchapi.dist</li>
    <li>python312Packages.submitit</li>
    <li>python312Packages.submitit.dist</li>
    <li>python312Packages.torchbench</li>
    <li>python312Packages.torchbench.dist</li>
  </ul>
</details>

**Review by GaetanLepage — APPROVED** [review]

**Pull request merged**